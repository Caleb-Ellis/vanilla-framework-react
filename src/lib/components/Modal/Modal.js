import React from 'react';
import PropTypes from 'prop-types';

const Modal = (props) => {
  const {
    children, onClick, show, title,
  } = props;

  return (
    <div
      className="p-modal"
      style={{ display: show ? 'flex' : 'none' }}
    >
      <div
        className="p-modal__dialog"
        role="dialog"
        aria-labelledby="modal-title"
        aria-describedby="modal-description"
      >
        <header className="p-modal__header">
          <h2 className="p-modal__title">{ title }</h2>
          <button
            className="p-modal__close"
            aria-label="Close active modal"
            onClick={onClick}
          >
            Close
          </button>
        </header>
        { children }
      </div>
    </div>
  );
};

Modal.defaultProps = {
  onClick: () => 1,
  show: false,
  title: null,
};

Modal.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  show: PropTypes.bool,
  title: PropTypes.string,
};

Modal.displayName = 'Modal';

export default Modal;
