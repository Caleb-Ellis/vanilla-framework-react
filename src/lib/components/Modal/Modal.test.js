import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Modal from './Modal';

describe('Modal component', () => {
  it('should be hidden if show prop is false', () => {
    const modal = ReactTestRenderer.create(
      <Modal>
        <p>Modal text</p>
      </Modal>);
    const json = modal.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should show if show prop is true', () => {
    const modal = ReactTestRenderer.create(
      <Modal show>
        <p>Modal text</p>
      </Modal>);
    const json = modal.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should display a title if prop is provided', () => {
    const modal = ReactTestRenderer.create(
      <Modal show title="Modal title">
        <p>Modal text</p>
      </Modal>);
    const json = modal.toJSON();
    expect(json).toMatchSnapshot();
  });
});
