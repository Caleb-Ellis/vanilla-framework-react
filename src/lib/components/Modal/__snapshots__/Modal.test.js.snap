// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should be hidden if show prop is false 1`] = `
<div
  className="p-modal"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    aria-describedby="modal-description"
    aria-labelledby="modal-title"
    className="p-modal__dialog"
    role="dialog"
  >
    <header
      className="p-modal__header"
    >
      <h2
        className="p-modal__title"
      />
      <button
        aria-label="Close active modal"
        className="p-modal__close"
        onClick={[Function]}
      >
        Close
      </button>
    </header>
    <p>
      Modal text
    </p>
  </div>
</div>
`;

exports[`Modal component should display a title if prop is provided 1`] = `
<div
  className="p-modal"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    aria-describedby="modal-description"
    aria-labelledby="modal-title"
    className="p-modal__dialog"
    role="dialog"
  >
    <header
      className="p-modal__header"
    >
      <h2
        className="p-modal__title"
      >
        Modal title
      </h2>
      <button
        aria-label="Close active modal"
        className="p-modal__close"
        onClick={[Function]}
      >
        Close
      </button>
    </header>
    <p>
      Modal text
    </p>
  </div>
</div>
`;

exports[`Modal component should show if show prop is true 1`] = `
<div
  className="p-modal"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    aria-describedby="modal-description"
    aria-labelledby="modal-title"
    className="p-modal__dialog"
    role="dialog"
  >
    <header
      className="p-modal__header"
    >
      <h2
        className="p-modal__title"
      />
      <button
        aria-label="Close active modal"
        className="p-modal__close"
        onClick={[Function]}
      >
        Close
      </button>
    </header>
    <p>
      Modal text
    </p>
  </div>
</div>
`;
